client/components/common/SideNavDrawer.tsx

```diff
@@ -31,21 +29,22 @@ export default function SideNavDrawer({
     animation: 'opacity-out 200ms ease-in',
     animationFillMode: 'forwards',
   };
-  const { data: userProfile, isError, error } = useFetchUserProfile();
-  const { showNotification } = useNotification();
 
-  if (isError) {
-    console.error('Failed to fetch user profile: ', error);
-    let errorMessage = 'Failed to fetch user profile. Please try again.';
-
-    if (error instanceof AxiosError && error.response) {
-      errorMessage = error.response.data || errorMessage;
-    } else if (error instanceof Error) {
-      errorMessage = error.message;
+  const { data: userProfile, status } = useFetchUserProfile();
+
+  const renderUserProfile = () => {
+    if (status === 'pending') {
+      return <Typography variant='caption'>Loading...</Typography>;
+    } else if (
+      status === 'success' &&
+      typeof userProfile !== 'string' &&
+      userProfile?.name
+    ) {
+      return <Typography variant='caption'>{userProfile.name}</Typography>;
```

Comment:
Do we need all three of these checks? I think this would also work:
```suggestion
    } else if (userProfile?.name) {
```

Reply:
Yes we will also need to check the status because useFetchUserProfile is asyc and there may be a delay getting the result. While it may seem lengthy to check both it will help prevent silent bugs down the line.

Reply:
Thanks for clearing this up. My thoughts were that if the status wasn't 'pending' and the query didn't throw, status === success would be implied and all we would need to check if the userProfile.name was present. At the end of the day, stricter checks as you have it seems like a good idea.